client/src/components/Upload.vue

```diff
@@ -5,15 +5,17 @@ import {
   sizeFormatter,
 } from '@girder/components/src/utils/mixins';
 import {
-  videoFilesRegEx,
-  imageFilesRegEx,
   ImageSequenceType,
   VideoType,
 } from '@/constants';
 import { makeViameFolder } from '@/lib/api/viame.service';
+import { mapActions, mapState } from 'vuex';
 import { getResponseError } from '@/lib/utils';
 
-function prepareFiles(files) {
+function prepareFiles(files, filetypes) {
+  const videoFilesRegEx = new RegExp(`${filetypes.video.join('$|')}$`, 'i');
+  const imageFilesRegEx = new RegExp(`${filetypes.image.join('$|')}$`, 'i');
```

Comment:
I see these lines repeated in several places.  Might be nice to just refactor them out into a Vuex Getter and get rid of the `mapState`.   See comment below.

Reply:
Oh yeah, I was looking at this earlier, but when I tried to make the RegEx's in the module, it threw an error cuz 'state' was possibly null?

Reply:
I don't know about state itself being null.  but the getter would need to handle the case where filetypes is null.